src/main/java/com/wolox/training/controllers/UserController.java

```diff
@@ -0,0 +1,165 @@
+package com.wolox.training.controllers;
+
+import com.wolox.training.exceptions.BookNotFoundException;
+import com.wolox.training.exceptions.UserNotFoundException;
+import com.wolox.training.exceptions.UserUsernameRepeatedException;
+import com.wolox.training.models.Book;
+import com.wolox.training.models.User;
+import com.wolox.training.repositories.BookRepository;
+import com.wolox.training.repositories.UserRepository;
+
+import io.swagger.v3.oas.annotations.Operation;
+import io.swagger.v3.oas.annotations.media.Content;
+import io.swagger.v3.oas.annotations.responses.ApiResponse;
+import io.swagger.v3.oas.annotations.responses.ApiResponses;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.stereotype.Controller;
+import org.springframework.web.bind.annotation.DeleteMapping;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+
+import java.util.Optional;
+
+@Controller
+@RequestMapping("/api/users")
+public class UserController {
+
+    @Autowired
+    private UserRepository userRepository;
+
+    @Autowired
+    private BookRepository bookRepository;
+
+    /**
+     * save the received user
+     * @param user user to be saved
+     * @exception UserUsernameRepeatedException when the username is already in the database.
+     * @return the user saved
+     */
+    @Operation(summary = "Create a user")
+    @ApiResponses(value = {
+            @ApiResponse(responseCode = "201", description = "Successfully created", content = @Content),
+            @ApiResponse(responseCode = "400", description = "User could not be created", content = @Content)
+    })
+    @PostMapping
+    public ResponseEntity<Object> create(@RequestBody User user) {
+        if(userRepository.findByUsername(user.getUsername()).isPresent()){
+            throw new UserUsernameRepeatedException();
+        }
+        return new ResponseEntity<>( userRepository.save(user), HttpStatus.CREATED);
+    }
+
+
+    /**
+     * delete user searched by id
+     * @param id to find user
+     * @exception UserNotFoundException when user was not found
+     * @return String is a descriptive text
+     */
+   @Operation(summary = "Delete a user")
+    @ApiResponses(value = {
+            @ApiResponse(responseCode = "201", description = "User was deleted successfully", content = @Content),
+            @ApiResponse(responseCode = "404", description = "User was not found", content = @Content)
+    })
+    @DeleteMapping("/{id}")
+    public ResponseEntity<String> delete(@PathVariable Long id) {
```

Comment:
The @PathVariable or @RequestParam can be documented more accurately using the @Parameter annotation

Here I leave you an example:
`@Parameter(name = "userId", description = "User id", example = "1", required = true)`